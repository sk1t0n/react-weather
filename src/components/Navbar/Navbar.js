import React from 'react';
import NavbarBootstrap from 'react-bootstrap/Navbar';
import { Container, Nav } from 'react-bootstrap';

function Navbar(props) {
  return (
    <NavbarBootstrap bg="dark" variant="dark" expand="sm">
      <Container>
        <NavbarBootstrap.Brand href="/">React-Weather</NavbarBootstrap.Brand>
        <NavbarBootstrap.Toggle aria-controls="basic-navbar-nav" />
        <NavbarBootstrap.Collapse id="basic-navbar-nav">
          <Nav className="me-auto">
            <Nav.Link href="/">Home</Nav.Link>
          </Nav>
        </NavbarBootstrap.Collapse>
      </Container>
    </NavbarBootstrap>
  );
}

export { Navbar };
