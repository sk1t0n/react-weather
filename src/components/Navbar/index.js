import React from 'react';
import NavbarBootstrap from 'react-bootstrap/Navbar';
import { Container, Nav } from 'react-bootstrap';
import { Link, useLocation } from 'react-router-dom';

const isActive = (location, path) => {
  console.log(location.pathname)
  if (path === '/' && location.pathname === '/') {
    return true;
  } else {
    return path.length > 1 && location.pathname.startsWith(path);
  }
}

function Navbar(props) {
  const location = useLocation();
  const links = [
    { title: 'Home', to: '/', active: isActive(location, '/') },
    { title: 'Info', to: '/info', active: isActive(location, '/info') }
  ];

  return (
    <NavbarBootstrap bg="dark" variant="dark" expand="sm">
      <Container>
        <NavbarBootstrap.Brand href="/">React-Weather</NavbarBootstrap.Brand>
        <NavbarBootstrap.Toggle aria-controls="basic-navbar-nav" />
        <NavbarBootstrap.Collapse id="basic-navbar-nav">
          <Nav className="me-auto">
            {links.map(link => (
              <Link 
                key={link.title}
                className={`nav-link ${link.active ? 'active' : ''}`}
                to={link.to}
              >
                { link.title }
              </Link>
            ))}
          </Nav>
        </NavbarBootstrap.Collapse>
      </Container>
    </NavbarBootstrap>
  );
}

export { Navbar };
