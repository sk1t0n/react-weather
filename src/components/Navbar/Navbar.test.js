import React from 'react';
import renderer from 'react-test-renderer';
import { Navbar } from './Navbar';

test('Render correctly navbar component', () => {
  const component = renderer.create(<Navbar />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
