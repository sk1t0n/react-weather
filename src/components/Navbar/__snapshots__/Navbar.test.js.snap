// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render correctly navbar component 1`] = `
<nav
  className="navbar navbar-expand-sm navbar-dark bg-dark"
>
  <div
    className="container"
  >
    <a
      className="navbar-brand"
      href="/"
    >
      React-Weather
    </a>
    <button
      aria-controls="basic-navbar-nav"
      aria-label="Toggle navigation"
      className="navbar-toggler collapsed"
      onClick={[Function]}
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      aria-expanded={null}
      className="navbar-collapse collapse"
      id="basic-navbar-nav"
    >
      <div
        className="me-auto navbar-nav"
        onKeyDown={[Function]}
      >
        <a
          className="nav-link"
          data-rr-ui-event-key="/"
          disabled={false}
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </div>
    </div>
  </div>
</nav>
`;
